### PR TITLE
fix: ELECTRON-1238 (Handle session permissions)

### PR DESCRIPTION
**Pull request by KiranNiranjan**
## Description
Handle session permissions
[ELECTRON-1238](https://perzoinc.atlassian.net/browse/ELECTRON-1238)

## Solution Approach
Verify global config's permissions field and prevent access to some properties

Notes: Add support for handling session permissions for media and others

### CODE REVIEW

File: src/app/window-actions.ts [hunk 1]
```diff
@@ -194,3 +207,65 @@ export const removeWindowEventListener = (window: BrowserWindow): void => {
...
+
+/**
+ * Sets permission requests for the window
+ *
+ * @param webContents {Electron.webContents}
+ */
+export const handlePermissionRequests = (webContents: Electron.webContents): void => {
+
+    const { session } = webContents;
+    if (!webContents || !session) {
```
Comment (keerthi16) [inline src/app/window-actions.ts]: `webContents` needs to be checked first

File: src/app/window-actions.ts [hunk 2]
```diff
@@ -194,3 +207,65 @@ export const removeWindowEventListener = (window: BrowserWindow): void => {
...
+ *
+ * @param permission {boolean} - config value to a specific permission
+ * @param message {string} - custom message displayed to the user
+ * @param callback {function}
+ */
+export const handleSessionPermissions = (permission: boolean, message: string, callback: (permission: boolean) => void): void => {
+    logger.info(`window-action: permission is ->`, { type: message, permission });
+
+    if (!permission) {
+        const fullMessage = `${i18n.t('Your administrator has disabled')()} ${message}. ${i18n.t('Please contact your admin for help')()}`;
```
Comment (VishwasShashidhar) [inline src/app/window-actions.ts]: we'll need a space after `disabled`

Comment (KiranNiranjan) [reply src/app/window-actions.ts]: @VishwasShashidhar There is already space after `disabled`

Comment (VishwasShashidhar) [reply src/app/window-actions.ts]: Ah! Templating, yes, thanks :)

**Pull request merged**